frontend/app/services/work-package-service.js

```diff
@@ -48,7 +48,11 @@ module.exports = function($http,
     if (workPackage.form) {
       _.forEach(workPackage.form.pendingChanges, function(value, field) {
         if (WorkPackageFieldService.isSpecified(workPackage, field)) {
-          if(field == 'date') {
+          if(field === 'date') {
+            if(WorkPackageFieldService.isMilestone(workPackage)) {
+              data['startDate'] = data['dueDate'] = value ? value : null;
```

Comment:
@myabc, It's checking for a empty string value. If value is empty string(which is not allowed by ISO date) then change it to null.